src/fprime_extras/lint/rules/rule_valid_port_types.py

```diff
@@ -0,0 +1,162 @@
+                        name = port.get('name')
+                        msg = 'Port "{}" specifies invalid type "{}". Imported types are {}.'
+                        msg = msg.format(name, port_type, imported_types)
+                        note = Notification(filename=extras_file.full_filename,
+                                            rule_slug=slug,
+                                            description=msg,
+                                            line_number=port.sourceline,
+                                            column=0,
+                                            severity=LintSeverity.CRITICAL
+                                            )
+                        notifications.append(note)
+        return notifications
+
+    def fix(self):
+        # if args.fix and flag:
+        #     conn.getparent().remove(conn)
+        pass
+
+    def reset(self):
+        pass
+
+    def get_imported_types(self, import_list, fprime_path, extras_file):
+        """
+        import_list: list of import_port_type etree elements
+        fprime_path: path to the fprime root
+        extras_file: file that imports the given files
+
+        parses the imported files and returns a hash set of their full type names as strings.
+        Will also return any critical notifications associated with parsing the imported types
+        """
+        imported_types = set()
+        notifications = []
+        for imp in import_list:
+            import_path = os.path.join(fprime_path, imp.text)
+            try:
+                xml_tree = etree.parse(import_path)
+                root = xml_tree.getroot()
+                if root.tag == "interface":
+                    name = root.get("name")
+                    namespace = root.get("namespace")
+                    full_name = namespace + "::" + name
+                    imported_types.add(full_name)
+                    log.info("Found port type: {}".format(full_name))
+                else:
+                    note = Notification(filename=extras_file.full_filename,
+                                        rule_slug=slug,
+                                        description="Component imported invalid port type: {}".format(imp.text),
+                                        line_number=imp.sourceline,
+                                        column=0,
+                                        severity=LintSeverity.CRITICAL
+                                        )
+                    notifications.append(note)
+            except XMLSyntaxError as e:
+                note = Notification(filename=extras_file.full_filename,
+                                    rule_slug=slug,
+                                    description=e.msg,
+                                    line_number=e.position[0],
+                                    column=e.position[1],
+                                    severity=LintSeverity.CRITICAL
+                                    )
```

Comment:
I see that this notification is formatted exactly as the Valid XML Document rule does, but if the line number and column are not for the original file, the reported notification will be very misleading...

Reply:
as in, it should be reporting the problem as the line number and column of the import statement, not the syntax error location of the other file.

Reply:
I fixed this to report the line number of the import causing the issue